storage_s3/indico_storage_s3/plugin.py

```diff
@@ -108,3 +111,31 @@ def send_file(self, file_id, content_type, filename, inline=True):
             return redirect(url)
         except Exception as e:
             raise StorageError('Could not send file "{}": {}'.format(file_id, e)), None, sys.exc_info()[2]
+
+    def get_bucket_name(self, name, replace_placeholders=True):
+        if replace_placeholders:
+            return self.replace_bucket_placeholders(name, datetime.datetime.now())
+        else:
+            return self.bucket
+
+    def create_bucket(self, name):
+        try:
+            response = self.client.get_bucket_acl(Bucket=self.acl_template_bucket)
+            acl_keys = ['Owner', 'Grants']
+            acl = {key: response[key] for key in response if key in acl_keys}
+            self.client.create_bucket(Bucket=name)
+            self.client.put_bucket_acl(AccessControlPolicy=acl, Bucket=name)
+        except Exception as e:
+            raise StorageError('Could not create bucket "{}": {}'.format(name, e)), None, sys.exc_info()[2]
+
+    def get_week_of_the_month(self, date):
+        first_day = date.replace(day=1)
+        current_day = date.day
+        return int(ceil((current_day + first_day.weekday())/7))
+
+    def replace_bucket_placeholders(self, name, date):
+        dates = {'year': str(date.year),
+                 'month': str(date.month),
+                 'week': str(self.get_week_of_the_month(date))}
+        pattern = re.compile("|".join(dates.keys()))
+        return pattern.sub(lambda m: dates[re.escape(m.group(0))], name)
```

Comment:
TBH, using a regex here seems strange. Why not simple `replace` calls?

```
name = name.replace('<year>', date.strftime('%Y'))
name = name.replace('<month>', date.strftime('%m'))
name = name.replace('<week>', date.strftime('%W'))  # or %U?
```

could also loop over a dict and call replace in the loop - but I think using a regex seems overkill and reduces readability a lot